src/darwin/Framework/CHIP/MTRControllerFactory.mm

```diff
@@ -446,6 +462,11 @@ - (MTRDeviceController * _Nullable)createController
         // Bringing up the first controller.  Start the event loop now.  If we
         // fail to bring it up, its cleanup will stop the event loop again.
         chip::DeviceLayer::PlatformMgrImpl().StartEventLoopTask();
+
+        if (_otaProviderDelegateBridge) {
+            auto systemState = _controllerFactory->GetSystemState();
+            _otaProviderDelegateBridge->Init(systemState->SystemLayer(), systemState->ExchangeMgr());
```

Comment:
Unfortunately, at this point there might not be a usable system state on the controller factory.  It gets created when the C++ controller is actually created, which has not happened here yet.  :(

And yes, I know I said it's about where we do the event loop bits.  I was wrong about this part here...

We really need to add some actual tests exercising this thing as controllers start up and shut down, to make sure it all works properly.

The right place to put this init is probably at the place where we start a controller successfully and the controller count is 1...